src/hotspot/share/prims/jvmtiEventController.cpp

```diff
@@ -1207,8 +1218,23 @@ JvmtiEventController::vm_init() {
 
 void
 JvmtiEventController::vm_death() {
+  // No new event callbacks except vm_death can be called after this point.
+  AtomicAccess::store(&_execution_finished, true);
   if (JvmtiEnvBase::environments_might_exist()) {
     MutexLocker mu(JvmtiThreadState_lock);
     JvmtiEventControllerPrivate::vm_death();
   }
-}
+
+  // Some events might be still in callback for daemons and VM internal threads.
+  const double start = os::elapsedTime();
+  const double max_wait_time = 60;
+  // The first time we see the callback count reach zero we know all actual
```

Comment:
Nit: s/the callback count reach zero/the callback count reaches zero/g

Reply:
The current text is correct in the tense used.